tests/Google/Ads/GoogleAds/Lib/V3/GoogleAdsClientBuilderTest.php

```diff
@@ -80,6 +82,15 @@ public function testBuildClientFromConfiguration()
         $this->assertSame(self::$LOGIN_CUSTOMER_ID, $googleAdsClient->getLoginCustomerId());
         $this->assertSame('https://abc.xyz:443', $googleAdsClient->getEndpoint());
         $this->assertSame('https://localhost:8080', $googleAdsClient->getProxy());
+        $this->assertSame(self::provideValidTransports()[0][0], $googleAdsClient->getTransport());
+    }
+
+    public function provideValidTransports()
```

Comment:
1. Is this needed to be public? Will `@dataProvider` work if this is private? (I know it would not be consistent with other methods, but we should do what is correct)
2. Can this be static? If not, using `$this->provideValidTransports` seems more semantically correct.

Reply:
1. Yes it does: see doc [here](https://phpunit.de/manual/3.7/en/writing-tests-for-phpunit.html#writing-tests-for-phpunit.data-providers)
1. Yes it can be static so I changed.